jedi/api/completion.py

```diff
@@ -33,7 +35,13 @@ def filter_names(evaluator, completion_names, stack, like_name):
         if settings.case_insensitive_completion:
             string = string.lower()
 
-        if string.startswith(like_name):
+        if settings.fuzzy_completion:
+            pattern = '.*'.join(like_name) + '.*'
+            match = re.match(pattern, string)
```

Comment:
This doesn't match `foobar` if you type `bar`. Is this intended or did you want to use `re.search` instead of `re.match`? Otherwise I like it very much. Please remove the last `.*` (it's not needed) and use `.*?` to use lazy matching, which I'm pretty sure is more efficient.

Also I would probably move the generation of the pattern to the top with `re.compile`.

Thanks for the PR!